File: data/CodeLlama/readable_format_HPEC/fortran/file247.f90
subroutine init_array(nr, nq, np, a, cFour)
implicit none

DATA_TYPE, dimension(np, nq, nr) :: a
DATA_TYPE, dimension(np, np) :: cFour
integer :: nr, nq, np
integer :: i, j, k

do i = 1, nr
do j = 1, nq
do k = 1, np
a(k, j, i) = ((DBLE(i - 1) * DBLE(j - 1)) + DBLE(k - 1)) / &
DBLE(np)
end do
end do
end do
do i = 1, np
do j = 1, np
cFour(j, i) = (DBLE(i - 1) * DBLE(j - 1)) / np
end do
end do
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file247.cpp
void init_array(int nr, int nq, int np, double a[][nq][nr], double cFour[][np])
{
 int i, j, k;
 for (i = 0; i < nr; i++)
 for (j = 0; j < nq; j++)
 for (k = 0; k < np; k++)
 a[k][j][i] = ((double)(i) * (double)(j) + (double)(k)) / (double)(np);
 for (i = 0; i < np; i++)
 for (j = 0; j < np; j++)
 cFour[j][i] = ((double)(i) * (double)(j)) / np;
}
